PcapDotNet/src/PcapDotNet.Packets/Dhcp/DhcpLayer.cs

```diff
@@ -0,0 +1,188 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using PcapDotNet.Base;
+using PcapDotNet.Packets.Arp;
+using PcapDotNet.Packets.Dhcp.Options;
+using PcapDotNet.Packets.Ethernet;
+using PcapDotNet.Packets.IpV4;
+
+namespace PcapDotNet.Packets.Dhcp
+{
+    /// <summary>
+    /// Represents a DHCP layer.
+    /// <seealso cref="DhcpDatagram"/>
+    /// </summary>
+    public sealed class DhcpLayer : SimpleLayer, IEquatable<DhcpLayer>
+    {
+        /// <summary>
+        /// Message op code
+        /// </summary>
+        public DhcpMessageType MessageType { get; set; }
+
+        /// <summary>
+        /// Hardware address type
+        /// </summary>
+        public ArpHardwareType HardwareType { get; set; }
+
+        /// <summary>
+        /// Hardware address length
+        /// </summary>
+        public byte HardwareAddressLength { get; set; }
```

Comment:
I see, makes sense.
I didn't realize the Hardware Address field has constant size, regardless of the Hardware Address Length field.
It makes sense to set it to 6 (you can use MacAddress.SizeOf instead of 6) when the helping property ClientMacAddress is used, just make sure it's clear from the property comment.